mysqlnd_azure.c

```diff
@@ -397,6 +446,9 @@ MYSQLND_METHOD(mysqlnd_azure_data, connect)(MYSQLND_CONN_DATA ** pconn,
                     SET_CLIENT_ERROR(conn->error_info, MYSQLND_AZURE_ENFORCE_REDIRECT_ERROR_NO, UNKNOWN_SQLSTATE, "Connection aborted because init redirection failed.");
```

Comment:
Same here for error message. Here is an error message when mysqlnd_azure.enableRedirect=on

Reply:
Put those logs inside the if/else